api/novelist.py

```diff
@@ -433,6 +459,58 @@ def get_recommendations(self, metadata, recommendations_info):
                 metadata.recommendations += self._extract_isbns(book_info)
         return metadata
 
+    def put_isbns_novelist(self, library):
+        collectionList = []
+        for c in library.collections:
+            collectionList.append(c.id)
+
+        i1 = aliased(Identifier)
+        i2 = aliased(Identifier)
+        e = aliased(Equivalency)
+        lp = aliased(LicensePool)
+
+        isbnQuery = select(
+            [i2.identifier]
+        ).select_from(
+            join(lp, i1, i1.id==lp.id)
+            .join(e, i1.id==e.input_id)
+            .join(i2, e.output_id==i2.id)
+        ).where(
+            and_(lp.collection_id.in_(collectionList), i2.type=="ISBN")
+        ).alias('lp_isbns')
```

Comment:
This will miss ISBNs that are the primary identifier for their LicensePool, rather associated with some proprietary identifier that is associated with an ISBN through some Equivalency. Right now Plympton is the only distributor that uses ISBNs as its primary identifier, but I'd prefer to get this working now and not have to come back to it later.

There are two ways of doing this. You can do two different queries, one of which doesn't have the join against `Equivalency`, or you can do an `outerjoin` against `Equivalency` instead of a regular join, and match on `or_(i1.type=="ISBN", i2.type=="ISBN")`.

Either way, you would test this by creating a LicensePool directly associated with an ISBN and making sure this method picked it up.

Reply:
This code is also missing a test.